src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4764,41 +4779,19 @@ void handleInitializer(IOperation innerInitializer)
-                    case OperationKind.DynamicInvocation:
-                        // PROTOTYPE(dataflow): Dynamic invocation needs adjustment in the OperationFactory to ensure that
-                        // dynamic member references are generated appropriately.
-                        AddStatement(Visit(innerInitializer));
-                        return;
-
-                    case OperationKind.Increment:
-                        // PROTOTYPE(dataflow): It looks like we can get here with an increment
-                        //                      See Microsoft.CodeAnalysis.CSharp.UnitTests.SemanticModelGetSemanticInfoTests.ObjectInitializer_InvalidElementInitializer_IdentifierNameSyntax
-                        //                      Just drop it for now to enable other test scenarios.
-                        return;
-
-                    case OperationKind.Literal:
-                        // PROTOTYPE(dataflow): See Microsoft.CodeAnalysis.VisualBasic.UnitTests.BindingErrorTests.BC30994ERR_AggrInitInvalidForObject
-                        //                      Just drop it for now to enable other test scenarios.
-                        return;
-
-                    case OperationKind.LocalReference:
-                        // PROTOTYPE(dataflow): See Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.IOperationTests.TestClone
-                        //                      Just drop it for now to enable other test scenarios.
-                        return;
-
-                    case OperationKind.BinaryOperator:
-                    case OperationKind.FieldReference:
-                        // PROTOTYPE(dataflow): See Microsoft.CodeAnalysis.VisualBasic.UnitTests.SimpleFlowTests.LogicalExpressionInErroneousObjectInitializer
-                        //                      Just drop it for now to enable other test scenarios.
-                        return;
+                    default:
+                        // This assert is to document the list of things we know are possible to go through the default handler. It's possible there
+                        // are other nodes that will go through here, and if a new test triggers this assert, it will likely be fine to just add
+                        // the node type to the assert. It's here merely to ensure that we think about whether that node type actually does need
+                        // special handling in the context of a collection or object initializer before just assuming that it's fine.
+#if DEBUG
+                        var validKinds = ImmutableArray.Create(OperationKind.Invocation, OperationKind.DynamicInvocation, OperationKind.Increment, OperationKind.Literal,
+                                                               OperationKind.LocalReference, OperationKind.BinaryOperator, OperationKind.FieldReference, OperationKind.Invalid);
+                        Debug.Assert(validKinds.Contains(innerInitializer.Kind));
```

Comment:
>validKinds.Contains(innerInitializer.Kind) [](start = 37, length = 42)

I am afraid this assert might be too noisy. If BinaryOperator can be there, then it feels like pretty much anything can be there in an error case.

Reply:
That's why I have the comment above. I just want to make sure that we are considering any new scenarios we find, just in case we add a new valid scenario that will need special handling in future language updates.

---
In reply to: [193582662](https://github.com/dotnet/roslyn/pull/27519#discussion_r193582662) [](ancestors = 193582662)

Reply:
> That's why I have the comment above. I just want to make sure that we are considering any new scenarios we find, just in case we add a new valid scenario that will need special handling in future language updates.

Might make it hard to dogfood debug build though. All kinds of things happen when you are typing code, things are incomplete.

---
In reply to: [193583897](https://github.com/dotnet/roslyn/pull/27519#discussion_r193583897) [](ancestors = 193583897,193582662)

Reply:
I'll try a slightly different assert and see if that works: OperationKind.(Dynamic)Invocation || innerInitializer.HasErrors().

---
In reply to: [193593292](https://github.com/dotnet/roslyn/pull/27519#discussion_r193593292) [](ancestors = 193593292,193583897,193582662)

Reply:
I attempted to modify this assert, but I still ended up needing to add a number of items back to it until it practically got back to this list.  @AlekseyTs unless you feel strongly that we should entirely remove this assert, I'm going to keep it as is.

---
In reply to: [193594390](https://github.com/dotnet/roslyn/pull/27519#discussion_r193594390) [](ancestors = 193594390,193593292,193583897,193582662)